src/SegmentedArray.chpl

```diff
@@ -519,6 +601,109 @@ module SegmentedArray {
       return hits;
     }
 
+    /*
+    Peel off one or more fields matching the regular expression, delimiter, from each string (similar
+    to string.partition), returning two new arrays of strings.
+    *Warning*: This function is experimental and not guaranteed to work.
+
+    Note: the regular expression engine used, re2, does not support lookahead/lookbehind
+
+    :arg delimter: regex delimter where the split in SegString will occur
+    :type delimter: string
+
+    :arg times: The number of times the delimiter is sought, i.e. skip over the first (times-1) delimiters
+    :type times: int
+
+    :arg includeDelimiter: If true, append the delimiter to the end of the first return array
+                            By default, it is prepended to the beginning of the second return array.
+    :type includeDelimiter: bool
+
+    :arg keepPartial: If true, a string that does not contain <times> instances of
+                      the delimiter will be returned in the first array. By default,
+                      such strings are returned in the second array.
+    :type keepPartial: bool
+
+    :arg left: If true, peel from the left
+    :type left: bool
+
+    :returns: Components to build 2 SegStrings (leftOffsets, leftVals, rightOffsets, rightVals)
+    */
+    proc peelRegex(const delimiter: string, const times: int, const includeDelimiter: bool, const keepPartial: bool, const left: bool) throws {
+      checkCompile(delimiter);
+      // should we do len check here? re2.compile('') is valid regex and matches everything
+      ref oa = offsets.a;
+      ref va = values.a;
+      const lengths = getLengths() - 1;
+      var leftEnd: [offsets.aD] int;
+      var rightStart: [offsets.aD] int;
+
+      forall (o, len, i) in zip(oa, lengths, offsets.aD) with (var myRegex = _unsafeCompileRegex(delimiter)) {
+        var matches = myRegex.matches(interpretAsString(va[o..#(len+1)]));
```

Comment:
So since I declared `var lengths = getLengths -1` here to match the set up from `peel` (which is different than what I did in `substringSearchRegex` where I did `var lengths = getLengths()`), I thought that I needed an extra `+1` to get the full string. 

But I just made with `interpretAsString(va[o..#(len+1)])` and `test_regex_peel` still passes. I guess none of the `peelRegex` functionality depends on whether the null byte is included on what is passed to `myRegex.matches`